modules/gateway/rpc.go

```diff
@@ -155,3 +154,50 @@ func (g *Gateway) Broadcast(name string, obj interface{}) {
 	}
 	wg.Wait()
 }
+
+// Broadcast calls an RPC on all of the peers in the Gateway's peer list. The
+// calls are run in parallel. Broadcasts are restricted to "one-way" RPCs,
+// which simply write an object and disconnect. This is why Broadcast takes an
+// interface{} instead of an RPCFunc.
+func (g *Gateway) Broadcast(name string, obj interface{}) {
+	peers := g.Peers()
+	g.broadcastTo(name, obj, peers)
+}
+
+// BroadcastMinVersion calls an RPC on all of the peers in the Gateway's peer
+// list that are at least the specified version. The calls are run in parallel.
+// Broadcasts are restricted to "one-way" RPCs, which simply write an object
+// and disconnect. This is why Broadcast takes an interface{} instead of an
+// RPCFunc. BroadcastMinVersion can be used to ensure new RPC calls are not
+// made to old nodes that wouldn't understand them.
+func (g *Gateway) BroadcastMinVersion(name string, obj interface{}, minVersion string) {
+	id := g.mu.RLock()
+	defer g.mu.RUnlock(id)
```

Comment:
Correct. This isn't much of a problem because `broadcastTo` does not need any locking of it's own. Are you concerned that the lock is held longer than needed, i.e. during the actual network writes? In that case yes, the lock is held longer than needed.

On the other hand, holding the lock for the entire broadcast ensures that broadcasts are guaranteed to execute in the order they are called.

But since this does not align with the behavior of the unversioned `Broadcast` method, I'll change it.